File: DataManagement/Sender.py
#!/usr/bin/python
'''
Script to send files back to Airglow
  -s to add SITE

History: 27 Sep 2012 - initial script written; Daniel J. Fisher (dfisher2@illinois.edu)
         12 Feb 2014 - Updated to v3.0 - txtcheck; Daniel J. Fisher (dfisher2@illinois.edu)

'''

# Import required modules
import os
import sys
from glob import glob

# Set up correct folder options
folder = {}
folder[0] = '/cygdrive/c/Sending/'
folder[1] = '/cygdrive/d/Sending/'
folder[2] = '/cygdrive/f/Sending/'
folder[3] = '/home/gps/Sending/'
folder[4] = '/home/airglow/Sending/'
folder[5] = '/home/scintmon/Sending/'
folder[6] = '/data/Sending/'
folder[7] = 'C:/Sending/'
folder[8] = 'D:/Sending/'
folder[9] = 'F:/Sending/'
destiny = 'tx@remote2.ece.illinois.edu:/rdata/airglow/rx/.'
mfs = 70 #minimum file size to send

# Go through all possible folders for data
files = []
for f in range(len(folder)):
    files = files + glob(folder[f]+'*.tar.gz*')
    files = files + glob(folder[f]+'*.txt')

# Send all files one by one; remove if sent successfully
for f in [fx for fx in files if os.stat(fx).st_size > mfs]:
    os.system('chmod 774 '+f)
    flag = os.system('scp ' + f + ' ' + destiny)
    if flag == 0:
        os.remove(f)
        print 'Completed...'

print 'All Sending is now Complete!'

File: DataManagement/Sender.pl
#!/usr/bin/perl -w

# Script to Send all tar.gz to the Rx Airglow Folder, Needs 3 Letter Site Abbr.
# Daniel Fisher
# 29 August 2011

use File::Basename;

## Code to stop if alread running this script
#chomp(my $result = `pidof -x Sender.pl`);
#my @result = split(/ /,$result);
#my $processes = scalar(@result);
#print "$processes\n";
#
#if($processes gt 1) {
#   exit;
#}

# Get in Proper Sending Directory
my $site = $ARGV[0];
print("-$site-\n");
if($site eq 'CAR') {
   $local = sprintf("/cygdrive/f/Sending");
} elsif($site eq 'NSO') {
   $local = sprintf("/data/Sending");
} elsif($site eq 'CAJ_P') {
   $local = sprintf("/data/Sending");
} elsif($site eq 'CAJ_S') {
   $local = sprintf("/home/scintmon/Sending");
} elsif($site eq 'CAJ_D') {
   $local = sprintf("/home/gps/Sending");
} elsif($site eq 'CTO') {
   $local = sprintf("/home/airglow/Sending");
} elsif($site eq 'SGT_P') {
   $local = sprintf("/home/airglow/Sending");
} elsif($site eq 'SGT_T') {
   $local = sprintf("/home/gps/Sending");
} elsif($site eq 'COL') {
   $local = sprintf("/home/gps/Sending");
} elsif($site eq 'CTO_L') {
   $local = sprintf("/home/gps/Sending");
} elsif($site eq 'UAO_Cloud') {
   $local = sprintf("/cygdrive/d/Sending");
} else {
   $local = sprintf("/cygdrive/c/Sending");
} 

my $dest = sprintf('tx@remote2.ece.illinois.edu:/rdata/airglow/rx');
chdir($local);

# Send All MiniME files
@files = `ls TempL* Cloud* *.tar.gz*`;
foreach $file(@files) {
   chomp($file);
   system('chmod 777',"$file")
   my $result = system('scp', "$file", "$dest");
   if($result eq 0) {
      system("rm -f $file");
      print("$file Sent Successfully\n");
   }
}

print "Complete!\n";

